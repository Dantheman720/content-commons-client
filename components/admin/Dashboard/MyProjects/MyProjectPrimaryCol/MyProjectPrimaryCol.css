.myProjects_actions {
  margin-right: 15px; }
  @media screen and (max-width: 767px) {
    .myProjects_actions {
      margin-right: 0; } }

.myProjects_thumbnail {
  flex: 1;
  margin-right: 15px; }
  @media screen and (max-width: 767px) {
    .myProjects_thumbnail {
      margin-right: 0;
      padding: 0 3px;
      max-width: 66px; } }
  .myProjects_thumbnail img {
    width: 100%; }

.myProjects_data {
  position: relative;
  flex: 2;
  display: flex;
  flex-direction: column; }
  .myProjects_data_title {
    flex: 1;
    margin-bottom: 0;
    font-size: 18px;
    font-weight: bold;
    color: #112e51; }
    @media screen and (max-width: 767px) {
      .myProjects_data_title {
        max-width: 155px;
        font-size: 14px; } }
    .myProjects_data_title:hover {
      color: #0071bc; }
  .myProjects_data_actions button {
    -webkit-appearance: none;
    padding: 0;
    border: none;
    background-color: transparent; }
  @media screen and (max-width: 767px) {
    .myProjects_data_actions_wrapper {
      display: none;
      position: absolute;
      top: 0;
      right: 8px;
      z-index: 2;
      padding: 10px;
      background-color: #fff;
      box-shadow: 1px 1px 1px 1px #b2b4b8; }
      .myProjects_data_actions_wrapper > span {
        display: none; } }
  .myProjects_data_actions_action {
    font-size: 11px;
    font-weight: normal; }
    @media screen and (max-width: 767px) {
      .myProjects_data_actions_action {
        display: block;
        padding: 3px;
        margin-bottom: 8px;
        font-size: 1em; }
        .myProjects_data_actions_action:not(:first-child) {
          padding: 3px !important; } }
  .myProjects_data_actions_mobileToggle {
    position: absolute;
    top: 0;
    right: 5px;
    font-size: 1.5em;
    color: #112e51; }
    @media screen and (min-width: 768px) {
      .myProjects_data_actions_mobileToggle {
        display: none; } }
    .myProjects_data_actions_mobileToggle i.icon {
      margin: 0;
      width: auto; }

td.items_table_item.displayDataActions {
  z-index: 3 !important; }
  td.items_table_item.displayDataActions .myProjects_data_actions_wrapper {
    display: block; }

.ui.popup.detailsFiles_popup {
  max-width: inherit;
  padding-top: 1em;
  padding-bottom: 1em;
  color: #5b6168;
  background-color: #f1f1f1; }
  @media screen and (max-width: 400px) {
    .ui.popup.detailsFiles_popup {
      margin-left: 1em; } }
  .ui.popup.detailsFiles_popup:before {
    background-color: #f1f1f1 !important; }
  .ui.popup.detailsFiles_popup p {
    margin-bottom: 3px;
    font-size: 11px; }
