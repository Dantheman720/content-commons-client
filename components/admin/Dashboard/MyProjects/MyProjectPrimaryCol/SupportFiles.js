import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import ApolloError from 'components/errors/ApolloError';
import { formatBytes } from 'lib/utils';

const VIDEO_PROJECT_SUPPORT_FILES_QUERY = gql`
  query VideoProjectSupportFiles( $id: ID! ) {
    videoProject( id: $id ) {
      id
      supportFiles {
        id
        url
        filetype
        filesize
        language {
          displayName
        }
        use {
          id
          name
        }
      }
    }
  }
`;

const SupportFiles = props => (
  <Query query={ VIDEO_PROJECT_SUPPORT_FILES_QUERY } variables={ { id: props.id } }>
    { ( { loading, error, data } ) => {
      if ( loading ) return <p>Loading....</p>;
      if ( error ) return <ApolloError error={ error } />;
      const { supportFiles } = data.videoProject;
      return (
        <div>
          <p>Files:</p>
          {
            supportFiles.map( file => {
              const {
                id,
                url,
                filetype,
                filesize,
                language: { displayName },
                use
              } = file;
              return (
                <p key={ id }>
                  { use.name } | <a href={ url }>{ filetype }</a> | <a href={ url }>{ displayName } { formatBytes( filesize ) }</a>
                </p>
              );
            } )
          }
        </div>
      );
    } }
  </Query>
);

SupportFiles.propTypes = {
  id: PropTypes.string
};

export default SupportFiles;
