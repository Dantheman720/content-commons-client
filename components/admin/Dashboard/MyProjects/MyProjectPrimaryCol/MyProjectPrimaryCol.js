/**
 *
 * MyProjectPrimaryCol
 *
 */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import truncate from 'lodash/truncate';
import {
  Checkbox, Icon, Modal, Popup
} from 'semantic-ui-react';
import VisuallyHidden from 'components/VisuallyHidden/VisuallyHidden';
import DetailsPopup from '../DetailsPopup/DetailsPopup';
import PreviewProjectItem from '../PreviewProjectItem';
import './MyProjectPrimaryCol.scss';

const handleDataActionsOffClick = e => {
  // Check if click target is a data actions menu link
  const isDataActionsMenuLink = e.target.classList.contains( 'myProjects_data_actions_action' );
  if ( isDataActionsMenuLink ) return;

  // If click target is not actions menu toggle button
  if ( !e.target.classList.contains( 'ellipsis' ) ) {
    // Close any open menus
    const openDataActionsMenu = document.querySelector( '.displayDataActions' );
    if ( openDataActionsMenu !== null ) openDataActionsMenu.classList.remove( 'displayDataActions' );

    // Remove the document click event handler since no open menus
    document.removeEventListener( 'click', handleDataActionsOffClick );
  }
};

const toggleDataActions = e => {
  e.persist();

  // Select target table td element
  const parentTD = e.target.closest( '.items_table_item' );

  if ( parentTD.classList.contains( 'displayDataActions' ) ) {
    parentTD.classList.remove( 'displayDataActions' );
  } else {
    // Close any other open dataActions menus
    const openDataActionsMenu = document.querySelector( '.displayDataActions' );
    if ( openDataActionsMenu !== null ) openDataActionsMenu.classList.remove( 'displayDataActions' );

    // Display target dataActions menu
    parentTD.classList.add( 'displayDataActions' );

    // Add a document click event listener to close any open data actions menus on 'off' clicks
    document.addEventListener( 'click', handleDataActionsOffClick );
  }
};

const MyProjectPrimaryCol = props => {
  const {
    d,
    d: { id },
    header,
    selectedItems,
    toggleItemSelection
  } = props;

  const projectTitleLength = d[header.name].length >= 35;
  return (
    <Fragment>
      <div className="myProjects_actions">
        <Checkbox
          data-label={ id }
          checked={ !!selectedItems.get( `${id}` ) }
          onChange={ toggleItemSelection }
        />
        { /* <div className="myProjects_favorite"><Icon name='star' /></div> */ }
      </div>
      <div className="myProjects_thumbnail">
        <img src={ d.thumbnail.url } alt={ d.thumbnail.alt } />
      </div>
      <div className="myProjects_data">
        { /**
           * Display tooltip for keyboard accessibility.
           * Tradeoff? Can't use <Link> in trigger and no
           * page transition animation, but err on side
           * of better accessibility.
           */ }
        { projectTitleLength
          ? (
            <Popup
              trigger={ (
                <a
                  href={ `/admin/project?conent=video&id=${id}&action=edit` }
                  className="myProjects_data_title"
                >
                  <span aria-hidden>{ truncate( d[header.name], { length: 35 } ) }</span>
                  <VisuallyHidden el="span">{ d[header.name] }</VisuallyHidden>
                </a>
              ) }
              content={ d[header.name] }
              hideOnScroll
              inverted
              on={ ['hover', 'focus'] }
              size="mini"
            />
          )
          : (
            <Link as={ `/admin/project/video/${id}/edit` } href={ `/admin/project?content=video&id=${id}&action=edit` }>
              <a className="myProjects_data_title">
                { d[header.name] }
              </a>
            </Link>
          ) }
        <div className="myProjects_data_actions">
          <div className="myProjects_data_actions_wrapper">
            <Link as={ `/admin/project/video/${id}/edit` } href={ `/admin/project?content=video&id=${id}&action=edit` }>
              <a className="linkStyle myProjects_data_actions_action">Edit</a>
            </Link>
            <span> | </span>
            <Modal
              trigger={ <button type="button" className="linkStyle myProjects_data_actions_action">Preview</button> }
              closeIcon
            >
              <Modal.Content>
                <PreviewProjectItem id={ id } />
              </Modal.Content>
            </Modal>
            <span> | </span>
            <DetailsPopup id={ id } />
          </div>
          <button
            type="button"
            className="linkStyle myProjects_data_actions_mobileToggle"
            onClick={ toggleDataActions }
          >
            <Icon name="ellipsis vertical" />
          </button>
        </div>
      </div>
    </Fragment>
  );
};

MyProjectPrimaryCol.propTypes = {
  d: PropTypes.object,
  header: PropTypes.object,
  selectedItems: PropTypes.instanceOf( Map ),
  toggleItemSelection: PropTypes.func
};

export default MyProjectPrimaryCol;
