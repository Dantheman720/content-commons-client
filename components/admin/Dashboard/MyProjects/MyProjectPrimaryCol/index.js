/**
 *
 * MyProjectPrimaryCol
 *
 */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import truncate from 'lodash/truncate';
import {
  Checkbox, Icon, Modal
} from 'semantic-ui-react';
import VisuallyHidden from 'components/VisuallyHidden/VisuallyHidden';
import DetailsPopup from '../DetailsPopup/DetailsPopup';
import ShareProjectItem from '../ShareProjectItem/ShareProjectItem';
import PreviewProjectItem from '../PreviewProjectItem';
import './MyProjectPrimaryCol.scss';

const handleDataActionsOffClick = e => {
  // Check if click target is a data actions menu link
  const isDataActionsMenuLink = e.target.classList.contains( 'myProjects_data_actions_action' );
  if ( isDataActionsMenuLink ) return;

  // If click target is not actions menu toggle button
  if ( !e.target.classList.contains( 'ellipsis' ) ) {
    // Close any open menus
    const openDataActionsMenu = document.querySelector( '.displayDataActions' );
    if ( openDataActionsMenu !== null ) openDataActionsMenu.classList.remove( 'displayDataActions' );

    // Remove the document click event handler since no open menus
    document.removeEventListener( 'click', handleDataActionsOffClick );
  }
};

const toggleDataActions = e => {
  e.persist();

  // Select target table td element
  const parentTD = e.target.closest( '.items_table_item' );

  if ( parentTD.classList.contains( 'displayDataActions' ) ) {
    parentTD.classList.remove( 'displayDataActions' );
  } else {
    // Close any other open dataActions menus
    const openDataActionsMenu = document.querySelector( '.displayDataActions' );
    if ( openDataActionsMenu !== null ) openDataActionsMenu.classList.remove( 'displayDataActions' );

    // Display target dataActions menu
    parentTD.classList.add( 'displayDataActions' );

    // Add a document click event listener to close any open data actions menus on 'off' clicks
    document.addEventListener( 'click', handleDataActionsOffClick );
  }
};

const MyProjectPrimaryCol = props => {
  const {
    d,
    d: { id },
    header,
    selectedItems,
    toggleItemSelection,
    windowWidth
  } = props;

  const projectTitleLength = d[header.name].length >= 35;
  return (
    <Fragment>
      <div className="myProjects_actions">
        <Checkbox
          data-label={ id }
          checked={ !!selectedItems.get( `${id}` ) }
          onChange={ toggleItemSelection }
        />
        { /* <div className="myProjects_favorite"><Icon name='star' /></div> */ }
      </div>
      <div className="myProjects_thumbnail">
        <img src={ d.thumbnail.url } alt={ d.thumbnail.alt } />
      </div>
      <div className="myProjects_data">
        <Link href={ `/admin/project/video/${id}/review` }>
          <a
            className="myProjects_data_title"
            title={ d[header.name] }
          >
            <span aria-hidden={ projectTitleLength }>{ truncate( d[header.name], { length: 35 } ) }</span>
            { projectTitleLength && <VisuallyHidden>{ d[header.name] }</VisuallyHidden> }
          </a>
        </Link>
        <div className="myProjects_data_actions">
          <div className="myProjects_data_actions_wrapper">
            <DetailsPopup id={ id } windowWidth={ windowWidth } />
            <span> | </span>
            <Modal
              trigger={ <button type="button" className="linkStyle myProjects_data_actions_action">Preview</button> }
              closeIcon
            >
              <Modal.Content>
                <PreviewProjectItem id={ id } />
              </Modal.Content>
            </Modal>
            <span> | </span>
            <ShareProjectItem id={ id } windowWidth={ windowWidth } />
          </div>
          <button
            type="button"
            className="linkStyle myProjects_data_actions_mobileToggle"
            onClick={ toggleDataActions }
          >
            <Icon name="ellipsis vertical" />
          </button>
        </div>
      </div>
    </Fragment>
  );
};

MyProjectPrimaryCol.propTypes = {
  d: PropTypes.object,
  header: PropTypes.object,
  selectedItems: PropTypes.instanceOf( Map ),
  toggleItemSelection: PropTypes.func,
  windowWidth: PropTypes.number
};

export default MyProjectPrimaryCol;
