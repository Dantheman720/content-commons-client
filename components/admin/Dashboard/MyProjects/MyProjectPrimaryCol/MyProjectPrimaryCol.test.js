import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Checkbox, Popup } from 'semantic-ui-react';
import MyProjectPrimaryCol from './MyProjectPrimaryCol';

/**
 * Need to mock Next.js dynamic imports
 * in order for this test suite to run.
 */
jest.mock( 'next-server/dynamic', () => () => 'VideoDetailsPopup' );
jest.mock( 'next-server/dynamic', () => () => 'ImageDetailsPopup' );

const props = {
  d: {
    id: '111',
    createdAt: 'May 9, 2019',
    updatedAt: 'May 9, 2019',
    projectTitle: 'Test Project',
    author: 'Jane Doe',
    status: 'PUBLISHED',
    visibility: 'PUBLIC',
    thumbnail: {
      alt: 'the alt text',
      url: 'https://website.com'
    }
  },
  header: {
    label: 'PROJECT TITLE',
    name: 'projectTitle'
  },
  selectedItems: new Map( [['999', true]] ),
  toggleItemSelection: jest.fn()
};

const Component = <MyProjectPrimaryCol { ...props } />;

describe( '<MyProjectPrimaryCol />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders an unchecked Checkbox if item is *not* selected', () => {
    const wrapper = shallow( Component );
    const checkbox = wrapper.find( Checkbox );

    expect( checkbox.prop( 'checked' ) ).toEqual( false );
  } );

  it( 'renders a checked Checkbox if item is selected', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { selectedItems: new Map( [[props.d.id, true]] ) } );

    const checkbox = wrapper.find( Checkbox );

    expect( checkbox.prop( 'checked' ) ).toEqual( true );
  } );

  it( 'checking/unchecking the Checkbox calls toggleItemSelection', () => {
    const wrapper = shallow( Component );
    const checkbox = wrapper.find( Checkbox );

    checkbox.simulate( 'change' );
    expect( props.toggleItemSelection ).toHaveBeenCalled();
  } );

  it( 'does not render a Popup if project title is less than 35 characters', () => {
    const wrapper = shallow( Component );
    const popup = wrapper.find( Popup );

    expect( props.d.projectTitle.length ).toBeLessThanOrEqual( 35 );
    expect( popup.exists() ).toEqual( false );
  } );

  it( 'renders a Popup with a truncated project title if it is over 35 characters', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( {
      d: {
        ...props.d,
        projectTitle: 'Test Project Test Project Test Project Test Project'
      }
    } );
    const popup = wrapper.find( Popup );

    expect( popup.exists() ).toEqual( true );
    expect( popup.prop( 'content' ) )
      .toEqual( 'Test Project Test Project Test Project Test Project' );
    expect( popup.prop( 'content' ).length ).toBeGreaterThan( 35 );
    expect( toJSON( popup ) ).toMatchSnapshot();
  } );

  it( 'renders "draft" class value for thumbnail img if status is DRAFT', () => {
    const wrapper = shallow( Component );
    const thumbnail = () => wrapper.find( '.myProjects_thumbnail img' );

    // initial value
    expect( thumbnail().prop( 'className' ) ).toEqual( null );

    wrapper.setProps( { d: { ...props.d, status: 'DRAFT' } } );
    expect( thumbnail().prop( 'className' ) ).toEqual( 'draft' );
  } );

  it( 'renders an overlay with "DRAFT" text for thumbnail img if status is DRAFT', () => {
    const wrapper = shallow( Component );
    const overlay = () => wrapper.find( '.draft-overlay' );

    // does not exist if status !== DRAFT
    expect( overlay().exists() ).toEqual( false );

    wrapper.setProps( { d: { ...props.d, status: 'DRAFT' } } );
    expect( overlay().exists() ).toEqual( true );
    expect( overlay().contains( <span>DRAFT</span> ) ).toEqual( true );
  } );
} );
