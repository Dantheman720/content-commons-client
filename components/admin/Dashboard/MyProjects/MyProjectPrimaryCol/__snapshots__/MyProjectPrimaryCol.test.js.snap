// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyProjectPrimaryCol /> renders a Popup with a truncated project title if it is over 35 characters 1`] = `
<Popup
  content="Test Project Test Project Test Project Test Project"
  hideOnScroll={true}
  inverted={true}
  keepInViewPort={true}
  on={
    Array [
      "hover",
      "focus",
    ]
  }
  position="top left"
  size="mini"
  trigger={
    <a
      className="projects_data_title"
      href="/admin/project/video/111/edit"
    >
      <span
        aria-hidden={true}
      >
        Test Project Test Project Test P...
      </span>
      <VisuallyHidden
        el="span"
      >
        Test Project Test Project Test Project Test Project
      </VisuallyHidden>
    </a>
  }
/>
`;

exports[`<MyProjectPrimaryCol /> renders without crashing 1`] = `
<Fragment>
  <div
    className="projects_actions"
  >
    <Checkbox
      checked={false}
      data-label="111"
      disabled={false}
      onChange={[MockFunction]}
      type="checkbox"
    />
  </div>
  <div
    className="projects_thumbnail"
  >
    <div
      className="wrapper"
    >
      <img
        alt="the alt text"
        className={null}
        src="https://website.com"
      />
    </div>
  </div>
  <div
    className="projects_data"
  >
    <Link
      as="/admin/project/video/111/edit"
      href="/admin/project?content=video&id=111&action=edit"
    >
      <a
        className="projects_data_title"
      >
        Test Project
      </a>
    </Link>
    <div
      className="projects_data_actions"
    >
      <div
        className="projects_data_actions_wrapper"
      >
        <Link
          as="/admin/project/video/111/edit"
          href="/admin/project?content=video&id=111&action=edit"
        >
          <a
            className="linkStyle projects_data_actions_action"
          >
            Edit
          </a>
        </Link>
        <span
          className="separator"
        >
          |
        </span>
        <Modal
          centered={true}
          closeIcon={true}
          closeOnDimmerClick={true}
          closeOnDocumentClick={false}
          dimmer={true}
          eventPool="Modal"
          trigger={
            <button
              className="linkStyle projects_data_actions_action"
              type="button"
            >
              Preview
            </button>
          }
        >
          <ModalContent>
            <Apollo(ProjectPreviewContent)
              id="111"
            />
          </ModalContent>
        </Modal>
        <span
          className="separator"
        >
          |
        </span>
        <DetailsPopup
          id="111"
        />
      </div>
      <button
        className="linkStyle projects_data_actions_mobileToggle"
        onClick={[Function]}
        type="button"
      >
        <Icon
          as="i"
          name="ellipsis vertical"
        />
      </button>
    </div>
  </div>
</Fragment>
`;
